vsphere/internal/helper/virtualmachine/virtual_machine_helper.go

```diff
@@ -278,6 +288,64 @@ func WaitForGuestNet(client *govmomi.Client, vm *object.VirtualMachine, routable
 	)
 	var v4gw, v6gw net.IP
 
+	log.Printf("[DEBUG] Expected ipv4 address [%s]", ipv4Addr)
+
+	if ipv4Addr != "" {
+		pp := client.PropertyCollector()
+		ctx, cancelpp := context.WithTimeout(context.Background(), time.Minute*time.Duration(timeout))
+		defer cancelpp()
+
+		err := property.Wait(ctx, pp, vm.Reference(), []string{"guest.net"}, func(pc []types.PropertyChange) bool {
+			for _, c := range pc {
+				if c.Op != types.PropertyChangeOpAssign {
+					continue
+				}
+
+				if c.Val == nil {
+					continue
+				}
+
+				var ip string = ""
+				switch v := c.Val.(type) {
+				case types.ArrayOfGuestNicInfo:
+					for _, n := range v.GuestNicInfo {
+						if n.IpConfig != nil {
+							for _, addr := range n.IpConfig.IpAddress {
+								ip = addr.IpAddress
+								parsedIP := net.ParseIP(addr.IpAddress)
+								if parsedIP.To4() == nil {
+									log.Printf("[DEBUG] Received IPv6. Skipping")
+									continue
+								} else {
+									break
+								}
+							}
+						}
+					}
+				}
+
+				log.Printf("[DEBUG] Received Ip address [%s]", ip)
+				if ip != "" {
+					if ip == ipv4Addr {
+						log.Printf("[DEBUG] Input ip: [%s], expected value: [%s]", ip, ipv4Addr)
```

Comment:
should log this outside if block. will help in debugging. Here it is anyways going to be same.

Reply:
Done